pkg/placement/placement.go

```diff
@@ -53,33 +52,53 @@ const (
 //  3. for updates on label selectors, re-evaluate if existing objects should be removed
 //     from clusters.
 func (c *Controller) handlePlacement(obj runtime.Object) error {
-	placement := obj.DeepCopyObject()
+	placement, err := runtimeObjectToPlacement(obj.DeepCopyObject())
+	if err != nil {
+		return err
+	}
 
 	// handle requeing for changes in placement, excluding deletion
 	if !isBeingDeleted(obj) {
-		if err := c.requeueForPlacementChanges(); err != nil {
+		if err := c.requeueForPlacementChanges(placement); err != nil {
 			return err
 		}
+	} else {
+		c.placementResolver.DeleteResolution(placement.GetName())
 	}
 
 	if err := c.handlePlacementFinalizer(placement); err != nil {
 		return err
 	}
 
+	// TODO (maroon): this should be deleted when transport is ready
 	if err := c.cleanUpObjectsNoLongerMatching(placement); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (c *Controller) requeueForPlacementChanges() error {
+func (c *Controller) requeueForPlacementChanges(placement *v1alpha1.Placement) error {
+	// update placement resolution destinations since placement was updated
+	list, err := ocm.ListClustersBySelectors(c.ocmClient, placement.Spec.ClusterSelectors)
+	if err != nil {
+		return err
+	}
+
+	c.placementResolver.UpdateDestinations(placement.Name, list)
+	if err := c.queuePlacementDecision(placement.GetName()); err != nil {
+		return fmt.Errorf("failed to requeue placement changes, failed to queue placement-decision: %w", err)
+	}
+
 	// allow some time before checking to settle
 	now := time.Now()
 	if now.Sub(c.initializedTs) < waitBeforeTrackingPlacements {
 		return nil
 	}
-	if err := c.requeueAll(); err != nil {
+
+	// requeue all objects to account for changes in placement.
+	// this does not include placement/placement-decision objects.
+	if err := c.requeueAllObjects(); err != nil {
 		return err
 	}
 	return nil
```

Comment:
This could be simpler.

```suggestion
	return c.requeueAllObjects()
```

Reply:
I originally left this and a few other similar instances for error wrapping later, but now changing all.